doc/nrf/templates/library_template_README.rst

```diff
@@ -0,0 +1,137 @@
+.. _library_template:
+
+Template: library
+#################
+
+.. contents::
+   :local:
+   :depth: 2
+
+.. note::
+   Use this template to create pages that describe the libraries included in |NCS|, following these instructions:
+
+   * Use this first section to give a very short description (1 or 2 sentences) of the library, indicating what it does and why it should be used.
+   * Give the library a concise name that also corresponds to the folder name.
+     If the library targets a specific protocol or device, add it in the title before the library name (for example, "NFC:" or "nRF9160:").
+     Do not include the word "library" in the title.
+     Use the provided stock phrases and includes when possible.
+   * Sections with ``*`` are optional and can be left out.
+     All the other sections are required for all libraries.
+     Do not add new sections or remove a mandatory section without consulting a tech writer.
+     If any new section is needed, try first to fit it as a subsection of one of the sections listed below.
+   * Include a tech writer in the PR review.
+
+Overview
+********
+
+.. note::
+   Use this section to give a general overview of the library.
+
+Implementation
+==============
+
+.. note::
+   Use this section to describe the library architecture and implementation.
+
+Supported features
+==================
+
+.. note::
+   Use this section to describe the features supported by the library.
+
+Supported backends*
+===================
+
+.. note::
+   Use this section to describe the backends supported by the library, if applicable.
+   This is optional.
+
+Requirements
+************
+
+.. note::
+   Use this section to list the requirements needed to use the library.
+   This is optional.
+
+Library files
+*************
+
+.. note::
+   Use this section to indicate the location of the library files.
+
+This library can be found under :file:`path_to_library_files` in the |NCS| folder structure.
+
+Configuration*
```

Comment:
A lot of IoT/Cloud libraries seem to contain info about initializing and connecting. Where would that fit into our template?

Reply:
I wonder if these sections shouldn't just be part of "Usage", as subsection, as they seem to be very specific to specific types of libraries.